File: 2017-18/contest2/prog/junior/enclosure.java
import java.io.*;
import java.util.*;

public class enclosure {
	public static void main(String[] args) throws IOException {
		Scanner input = new Scanner(System.in);
		for(int n = 0; n < 5; ++n) { // read 5 input cases
			String exp = input.nextLine();
			ArrayList<Integer> res = new ArrayList<Integer>();
			if(exp.indexOf(")") < 0) { // has left missing right
				int beg = exp.indexOf("("); int end = exp.length();
				for(int i = beg+1; i < end; ++i) {
					char c = exp.charAt(i);
					if(c == '+' || c == '-' || c == '*' || c == '/')
						res.add(i+3);
				}
			} else if(exp.indexOf("(") < 0) { // has right missing left
				int beg = 0; int end = exp.indexOf(")");
				for(int i = beg; i < end; ++i) {
					char c = exp.charAt(i);
					if(c == '+' || c == '-' || c == '*' || c == '/')
						res.add(i);
				}
			}
			for(int i = 0; i < res.size(); ++i) {
				System.out.print(res.get(i));
				if(i != res.size()-1) {
					System.out.print(", ");
				}
			}
			System.out.println();
		}
	}
}

File: 2017-18/contest2/prog/junior/enclosure.py
#!/usr/bin/python3
import sys
import math

for n in range(0,5):
	exp = input().strip()
	res = []
	if ')' not in exp: # has left missing right
		for i in range(exp.find('('), len(exp)):
			if exp[i] == '+' or exp[i] == '-' or exp[i] == '*' or exp[i] == '/':
				res.append(i+3)
	elif '(' not in exp: # has right missing left
		for i in range(0, exp.find(')')):
			if exp[i] == '+' or exp[i] == '-' or exp[i] == '*' or exp[i] == '/':
				res.append(i)
	j = 0
	for num in res:
		print(num, end = ", " if j < len(res)-1 else '\n')
		j += 1
